.github/workflows/community_ci.yml

```diff
@@ -62,68 +30,148 @@ jobs:
+    timeout-minutes: 180 # 3 hours
+    permissions:
+      statuses: write
+    env:
+      GITHUB_HEAD_REF: ${{ github.head_ref }}
+      MAIN_BRANCH_NAME: "augustin/04-18-community-ci_workflow_prevent_injection"
+      # TODO
+      #MAIN_BRANCH_NAME: "master"
+
+    steps:
+      - name: Checkout fork
+        uses: actions/checkout@v4
+        with:
+          # This checkouts a fork which can contain untrusted code
+          # It's deemed safe as the community-ci environment requires manual reviewer approval to run
+          repository: ${{ github.event.pull_request.head.repo.full_name }}
+          ref: ${{ github.head_ref }}
+          fetch-depth: 1
+
+      # On checkout the head commit is a detached head
+      # We have to use the git CLI to get the last commit id of the head branch
+      # We also perform validation and sanitization on the GITHUB_HEAD_REF to prevent command injection
+      - name: Fetch last commit id from remote branch
+        id: fetch_last_commit_id_pr
+        run: |
+
+          # Validation: Check if GITHUB_HEAD_REF contains only alphanumeric characters, hyphens, underscores or slashes
+          if [[ ! "$GITHUB_HEAD_REF" =~ ^[a-zA-Z0-9_/-]+$ ]]; then
+            echo "Invalid input: '${GITHUB_HEAD_REF}'. Only alphanumeric characters, hyphens, underscores or slashes are allowed."
+            exit 1
+          fi
+
+          # Sanitization: Remove any characters that are not alphanumeric, hyphens, underscores, or slashes
+          sanitized_head_ref=$(echo "$GITHUB_HEAD_REF" | tr -cd '[:alnum:]-_/')
+          echo "commit_id=$(git ls-remote --heads origin '${sanitized_head_ref}' | cut -f 1)" >> $GITHUB_OUTPUT
+
+      # This will sync the .github folder of the main repo with the fork
+      # This allows us to use up to date actions from the main repo
+      - name: Pull .github folder from main repository
+        id: pull_github_folder
+        run: |
+          git remote add main https://github.com/airbytehq/airbyte.git
+          git fetch main ${MAIN_BRANCH_NAME}
+          git checkout main/${MAIN_BRANCH_NAME} -- .github
+
+      - name: Run airbyte-ci connectors test
+        # This path refers to the fork .github folder.
+        # We make sure its content is in sync with the main repo .github folder by pulling it in the previous step
         uses: ./.github/actions/run-airbyte-ci
         with:
+          # TODO remove this binary URL before merging
+          airbyte_ci_binary_url: https://storage.googleapis.com/dev-airbyte-cloud-connector-metadata-service/airbyte-ci/releases/ubuntu/3570293/airbyte-ci
           context: "pull_request"
-          dagger_cloud_token: ${{ secrets.DAGGER_CLOUD_TOKEN }}
+          dagger_cloud_token: ${{ secrets.DAGGER_CLOUD_TOKEN_2 }}
           docker_hub_password: ${{ secrets.DOCKER_HUB_PASSWORD }}
           docker_hub_username: ${{ secrets.DOCKER_HUB_USERNAME }}
           gcp_gsm_credentials: ${{ secrets.GCP_GSM_CREDENTIALS }}
```

Comment:
Should we in theory make a separate GSM credential for CI runs? Not sure if this adds any extra security, my logic was that in case something goes wrong, we can rotate one cred, not the main one.

Reply:
Yes it'd indeed be a better practice. Can this be a followup 🙏